core/processor/abci.go

```diff
@@ -677,22 +677,17 @@ func (app *App) OnCommit() (resp tmtypes.ResponseCommit) {
 	cpt, _ := app.checkpoint.Checkpoint(app.blockCtx, app.currentTimestamp)
 	t0 := time.Now()
 
-	var snapHash []byte
-	var err error
-	// if there is an approved protocol upgrade proposal and the current block height is later than the proposal's block height then take a snapshot and wait to be killed by the process manager
-	if app.protocolUpgradeService.TimeForUpgrade() {
-		snapHash, err = app.snapshot.SnapshotNow(app.blockCtx)
-		if err == nil {
-			app.protocolUpgradeService.SetReadyForUpgrade()
-		}
-	} else {
-		snapHash, err = app.snapshot.Snapshot(app.blockCtx)
-	}
-
+	snapHash, err := app.snapshot.Snapshot(app.blockCtx)
```

Comment:
I don't think this is a good change for a couple of reasons:

1. if it's not normally a snapshot block, it will not take a snapshot - we want it to and need it to. 
2. The regular snapshot call is done and saved asynchronously - in this case we want to wait for it to complete synchronously before telling the visor that we're done and ready to die

Reply:
Oh yeah, sorry I have missed that the `now` at the end. The reason why I had to change was because the previous code was leading to panic. Beacuase:
1. `TimeForUpgrade` for checking `currentBlock  == upgradeBlock`
2. But `SetReadyForUpgrade` has a check for `currentBlock - upgradeBlock == 1` otherwise panic.
I will fix it.